scm-ui/ui-webapp/src/repos/branches/components/AheadBehindTag.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020-present Cloudogu GmbH and Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { Branch, BranchDetails } from "@scm-manager/ui-types";
+import React, { FC } from "react";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import { Tooltip } from "@scm-manager/ui-components";
+
+type Props = {
+  branch: Branch;
+  details: BranchDetails;
+};
+
+type BarProps = {
+  width: number;
+  direction: "right" | "left";
+};
+
+const Ahead = styled.div`
+  border-left: 1px solid gray;
+`;
+
+const Behind = styled.div``;
+
+const Count = styled.div`
+  word-break: keep-all;
+`;
+
+const Bar = styled.span.attrs(props => ({}))<BarProps>`
+  height: 3px;
+  max-width: 100%;
+  margin-top: -2px;
+  margin-bottom: 2px;
+  width: ${props => props.width}%;
+  border-radius: ${props => (props.direction === "left" ? "25px 0 0 25px" : "0 25px 25px 0")};
+`;
+
+const AheadBehindTag: FC<Props> = ({ branch, details }) => {
+  const [t] = useTranslation("repos");
+
+  if (branch.defaultBranch) {
+    return null;
+  }
+
+  const calculateBarLength = (part: number) => {
+    let lengthInPercent = 5;
+
+    if (part < 10) {
+      lengthInPercent += part;
+    }
+    if (part < 50) {
+      lengthInPercent += part / 5;
+    }
+    if (part < 500) {
+      lengthInPercent += part / 10;
+    }
+    if (part > 500) {
+      lengthInPercent += part / 25;
+    }
+    return lengthInPercent;
+  };
```

Comment:
What do you think about this calculation?

Reply:
Indeed I thought about this a bit and played around with sqrt and log and so on, but could not find something where I had a 💡 moment.

What I would change is that I'd subtract the amount from `part` that already was considered like this, if this makes sense:

```java
    if (part < 10) {
      lengthInPercent += part;
    }
    if (part < 50) {
      lengthInPercent += (part - 10) / 5;
    }
    if (part < 500) {
      lengthInPercent += (part - 60) / 10;
    }
    if (part > 500) {
      lengthInPercent += (part - 560) / 25;
    }
```